marc_to_solr/lib/process_holdings_helpers.rb

```diff
@@ -0,0 +1,166 @@
+    holding['location'] ||= Traject::TranslationMap.new("locations", default: "__passthrough__")[holding['location_code']]
+    holding['library'] ||= Traject::TranslationMap.new("location_display", default: "__passthrough__")[holding['location_code']]
+    # calculate call_number for permanent location
+    holding['call_number'] ||= []
+    holding['call_number'] << [field_852['h'], field_852['i'], field_852['k'], field_852['j']].compact.reject(&:empty?)
+    holding['call_number'].flatten!
+    holding['call_number'] = holding['call_number'].join(' ').strip if holding['call_number'].present?
+    holding['call_number_browse'] ||= []
+    holding['call_number_browse'] << [field_852['h'], field_852['i'], field_852['k'], field_852['j']].compact.reject(&:empty?)
+    holding['call_number_browse'].flatten!
+    holding['call_number_browse'] = holding['call_number_browse'].join(' ').strip if holding['call_number_browse'].present?
+    if field_852['l']
+      holding['shelving_title'] ||= []
+      holding['shelving_title'] << field_852['l']
+    end
+    if field_852['z']
+      holding['location_note'] ||= []
+      holding['location_note'] << field_852['z']
+    end
+    holding
+  end
+
+  # Build the items array in all_holdings hash
+  def holding_items(value:, all_holdings:, item:)
+    if all_holdings[value].present?
+      if all_holdings[value]["items"].nil?
+        all_holdings[value]["items"] = [item]
+      else
+        all_holdings[value]["items"] << item
+      end
+    end
+    all_holdings
+  end
+
+  def build_item(item:, field_852:, field_876:)
+    is_scsb = scsb?(field_852)
+    item[:holding_id] = field_876['0'] if field_876['0']
+    item[:enumeration] = field_876['3'] if field_876['3']
+    item[:id] = field_876['a'] if field_876['a']
+    item[:status_at_load] = field_876['j'] if field_876['j']
+    item[:barcode] = field_876['p'] if field_876['p']
+    item[:copy_number] = field_876['t'] if field_876['t']
+    item[:use_statement] = field_876['h'] if field_876['h'] && is_scsb
+    item[:storage_location] = field_876['l'] if field_876['l'] && is_scsb
+    item[:cgd] = field_876['x'] if field_876['x'] && is_scsb
+    item[:collection_code] = field_876['z'] if field_876['z'] && is_scsb
+    item
+  end
+
+  def process_866_867_868_fields(fields:, all_holdings:, holding_id:)
+    fields.each do |field|
+      location_has_value = []
+      supplements_value = []
+      indexes_value = []
+      location_has_value << field['a'] if field.tag == '866' && field['a']
+      location_has_value << field['z'] if field.tag == '866' && field['z']
+      supplements_value << field['a'] if field.tag == '867' && field['a']
+      supplements_value << field['z'] if field.tag == '867' && field['z']
+      indexes_value << field['a'] if field.tag == '868' && field['a']
+      indexes_value << field['z'] if field.tag == '868' && field['z']
+      next unless all_holdings[holding_id]
```

Comment:
Could this be a guard clause at the top of the method (like `return all_holdings if all_holdings[holding_id].nil?`)